sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java

```diff
@@ -0,0 +1,307 @@
+      given()
+          .when()
+          .delete(
+              BASE_PATH + "/{document-id}",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(401);
+    }
+
+    @Test
+    public void testDelete() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .delete(
+              BASE_PATH + "/{document-id}",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(204);
+
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .param("raw", true)
+          .when()
+          .get(
+              BASE_PATH + "/{document-id}",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(404);
+    }
+
+    @Test
+    public void testDeleteNotFound() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .param("raw", true)
+          .when()
+          .get(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(404);
+
+      // When a delete occurs on an unknown document, it still returns 204 No Content
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .delete(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(204);
+
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .param("raw", true)
+          .when()
+          .get(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(404);
+    }
+  }
+
+  @Nested
+  class DeleteDocumentPath {
+
+    @Test
+    public void unauthorized() {
+      given()
+          .when()
+          .delete(
+              BASE_PATH + "/{document-id}/test",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(401);
+    }
+
+    @Test
+    public void testDeletePath() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .param("raw", true)
+          .when()
+          .get(
+              BASE_PATH + "/{document-id}/test",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(200);
+
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .delete(
+              BASE_PATH + "/{document-id}/test",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(204);
+
+      given()
```

Comment:
let's also get a whole document and ensure there is no `test`:

`.body("test", nullValue());`

Reply:
same for the array case below